### ISSUE TITLE
type mismatch in assignment error wording

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#11118]
I think the type mismatch in assigment error needs to be adjusted.

Consider
``` chapel
var a: int;
var b: string;

a = b;
```

The compiler currently outputs
```
zz.chpl:4: error: type mismatch in assignment from string to int(64)
```

This error consistently confuses me because the assigment with the error has `int` as the left-hand-side type and `string` on the right-hand-side type but the error message swaps the order (even if it's correct if you try to read and understand the words).

**Comment by mppf** [comment]
These all would be better for me:
 * `Cannot assign to int from string`
 * `type mismatch setting int from string`
 * `type mismatch in int = string`

**Comment by bradcray** [comment]
I like either your first option or "type mismatch in assignment from int(64) to string".  That said, yours is probably better given that there's no requirement that the LHS and RHS types match in an assignment.

**Comment by dungarpan** [comment]
@mppf I want to solve this issue. In which file should I make the changes?

**Comment by mppf** [comment]
hi @dungarpan - thanks for your interest in helping us with this issue.

I'd recommend using `git grep` to search for the error message yourself - `git grep 'type mismatch in assignment'`.

Two things to note:
 1. You don't need to necessarily change *all* errors of this form in one PR. In particular I'd focus on the ones in `compiler/` first.
 2. You should make sure that there is agreement on the exact wording you intend to use before you go very far.

So, why don't you start by showing exactly what you intend the error to be for the example program. Once you've done that, I'll ask around for additional feedback on the proposed improvement.

**Comment by krishnadey30** [comment]
@mppf I would like to take this up. 

In `compiler/resolution/functionResolution.cpp`
![changes](https://user-images.githubusercontent.com/22263782/54354808-8eabf080-467d-11e9-9612-cb031ef2b684.png)

![output](https://user-images.githubusercontent.com/22263782/54354657-3d036600-467d-11e9-8332-b5bce77372d0.png)

**Comment by mppf** [comment]
hey @krishnadey30 - good, the next step is to `git grep 'type mismatch in assignment'` and update the test .good files that have this pattern. Make sure they pass (`start_test test/some/path/to/file.chpl`).

**Comment by krishnadey30** [comment]
@mppf I will do that and let you know.

**Comment by krishnadey30** [comment]
@mppf In `test/users/kreider/feature_unusedcode.chpl` I didn't get any error but a `feature_unusedcode.good` file is present and its not passing the test.

![image](https://user-images.githubusercontent.com/22263782/54367218-09820500-4698-11e9-8b67-e269fd073502.png)

![image](https://user-images.githubusercontent.com/22263782/54367313-346c5900-4698-11e9-8334-2c95d1531b60.png)

**Comment by mppf** [comment]
@krishnadey30 - that's a future, don't worry about it. You should be concerned about test output that shows as a Failure (i.e. #failures is not 0)

**Comment by krishnadey30** [comment]
@mppf  I have completed the changes you suggested and have also checked all the test files. I will open a PR.

**Comment by krishnadey30** [comment]
@mppf  I have opened a PR https://github.com/chapel-lang/chapel/pull/12596 . If some more changes are needed then I can do them.

**Comment by mppf** [comment]
Closed by @krishnadey30's PR #12596.
